pkg/kubelet/cm/container_manager_linux.go

```diff
@@ -137,6 +137,8 @@ type containerManagerImpl struct {
 	deviceManager devicemanager.Manager
 	// Interface for CPU affinity management.
 	cpuManager cpumanager.Manager
+	// Internal lifecycle event handlers for container resource management.
+	internalLifecycle InternalContainerLifecycle
```

Comment:
Can you share more about your decision to create a new `internalLifecycle` field on the `containerManagerImpl` struct?

Am I correct that it's essentially caching the results of the calling `&internalContainerLifecycleImpl{cm.cpuManager}`?

Reply:
Yes, it is caching the results of the calling `&internalContainerLifecycleImpl{cm.cpuManager}`, because I need to trigger the `internalLifecycle` to `PostStopContainer ` to release the container's cpu and I do not find other way to do it.

Reply:
I still don't understand why this is necessary. It looks like internalLifecycle is just a stateless wrapper around the cpu manager and topology manager? I guess it doesn't hurt to cache it, but this is a smaller PR without this change.